R/time.R

```diff
@@ -0,0 +1,197 @@
+#' Time Feature Generator
+#'
+#' `step_time` creates a *specification* of a recipe
+#'  step that will convert date-time data into one or more factor or
+#'  numeric variables.
+#'
+#' @inheritParams step_pca
+#' @inheritParams step_center
+#' @param ... One or more selector functions to choose variables
+#'  for this step. The selected variables should have class
+#'  `POSIXct` or `POSIXlt`. See [selections()] for more details.
+#' @param features A character string that includes at least one
+#'  of the following values: `ampm` (AM or PM), `hour24`, `hour12`,
+#'  `minute`,`second`, `decimal_day`.
+#' @param columns A character string of variables that will be
+#'  used as inputs. This field is a placeholder and will be
+#'  populated once [prep()] is used.
+#' @param keep_original_cols A logical to keep the original variables in the
+#'  output. Defaults to `TRUE`.
+#' @template step-return
+#' @family dummy variable and encoding steps
+#' @export
+#' @details Unlike some other steps, `step_time` does *not*
+#'  remove the original time variables by default. Set `keep_original_cols`
+#'  to `FALSE` to remove them.
+#'
+#'  See [step_date()] if you want to calculate features that are larger than
+#'  hours.
+#'
+#'  # Tidying
+#'
+#'  When you [`tidy()`][tidy.recipe()] this step, a tibble with columns
+#'  `terms` (the selectors or variables selected) and `value` (the feature
+#'  names).
+#'
+#' @examples
+#' library(lubridate)
+#'
+#' examples <- data.frame(
+#'   times = ymd_hms("2022-05-06 10:01:07") +
+#'   hours(1:5) + minutes(1:5) + seconds(1:5)
+#' )
+#' time_rec <- recipe(~ times, examples) %>%
+#'   step_time(all_predictors())
+#'
+#' tidy(time_rec, number = 1)
+#'
+#' time_rec <- prep(time_rec, training = examples)
+#'
+#' time_values <- bake(time_rec, new_data = examples)
+#' time_values
+#'
+#' tidy(time_rec, number = 1)
+step_time <-
+  function(recipe,
+           ...,
+           role = "predictor",
+           trained = FALSE,
+           features = c("hour24", "minute", "second"),
```

Comment:
I'm somewhat partial to using just `"hour"` to mean `"hour24"`, as most people will understand that this is what it means. Then only `"hour12"` is special